src/test/java/com/google/cloud/teleport/templates/common/SplunkConvertersTest.java

```diff
@@ -309,6 +313,110 @@ public void testFailsafeStringToSplunkEventValidIndex() {
     pipeline.run();
   }
 
+  /** Test successful conversion of JSON messages with user provided index 'fields'. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testFailsafeStringToSplunkEventValidFields() {
+
+    FailsafeElement<String, String> input =
+        FailsafeElement.of(
+            "",
+            "{\n"
+                + "\t\"name\": \"Jim\",\n"
+                + "\t\"_metadata\": {\"fields\":{\"test-key\":\"test-value\"}}\n"
+                + "}");
+
+    pipeline.getCoderRegistry().registerCoderForClass(SplunkEvent.class, SplunkEventCoder.of());
+
+    PCollectionTuple tuple =
+        pipeline
+            .apply(
+                Create.of(input)
+                    .withCoder(FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())))
+            .apply(
+                SplunkConverters.failsafeStringToSplunkEvent(
+                    SPLUNK_EVENT_OUT, SPLUNK_EVENT_DEADLETTER_OUT));
+
+    PAssert.that(tuple.get(SPLUNK_EVENT_DEADLETTER_OUT)).empty();
+    PAssert.that(tuple.get(SPLUNK_EVENT_OUT))
+        .containsInAnyOrder(
+            SplunkEvent.newBuilder()
+                .withEvent("{\"name\":\"Jim\"}")
+                .withFields(GSON.fromJson("{\"test-key\":\"test-value\"}", JsonObject.class))
+                .build());
+
+    pipeline.run();
+  }
+  
+  /** Test successful conversion of JSON messages with user provided empty index 'fields'. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testFailsafeStringToSplunkEventEmptyFields() {
+    
+    FailsafeElement<String, String> input =
+        FailsafeElement.of(
+            "",
+            "{\n"
+                + "\t\"name\": \"Jim\",\n"
+                + "\t\"_metadata\": {\"fields\":{}}\n"
+                + "}");
+    
+    pipeline.getCoderRegistry().registerCoderForClass(SplunkEvent.class, SplunkEventCoder.of());
+    
+    PCollectionTuple tuple =
+        pipeline
+            .apply(
+                Create.of(input)
+                    .withCoder(FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())))
+            .apply(
+                SplunkConverters.failsafeStringToSplunkEvent(
+                    SPLUNK_EVENT_OUT, SPLUNK_EVENT_DEADLETTER_OUT));
+    
+    PAssert.that(tuple.get(SPLUNK_EVENT_DEADLETTER_OUT)).empty();
+    PAssert.that(tuple.get(SPLUNK_EVENT_OUT))
+        .containsInAnyOrder(
+            SplunkEvent.newBuilder()
+                .withEvent("{\"name\":\"Jim\"}")
+                .withFields(GSON.fromJson("{}", JsonObject.class))
+                .build());
+    
+    pipeline.run();
+  }
+  
+  /** Test successful conversion of JSON messages with user provided invalid index 'fields'. */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testFailsafeStringToSplunkEventInvalidFields() {
```

Comment:
Better method name will be testFailsafeStringToSplunkEvent_withInValidFields

Reply:
Changed to `testFailsafeStringToSplunkEvent_withInvalidFields`